spark-statsd-reporter/pom.xml

```diff
@@ -11,14 +11,42 @@
   <properties>
     <maven.compiler.source>1.8</maven.compiler.source>
     <maven.compiler.target>1.8</maven.compiler.target>
-    <dropwizard.metrics.version>3.1.5</dropwizard.metrics.version>
+    <codahale.metrics.version>4.0.5</codahale.metrics.version>
+    <dropwizard.metrics.version>1.3.10</dropwizard.metrics.version>
```

Comment:
do we need it for `BaseReporterFactory` only? if so, please see my comment below